### PR TITLE
Elves Research

### LINKED ISSUE #13: Research blocking
Add structures and furniture behind research wall.

### PR DESCRIPTION
**Pull request by RodEsp**
This is a work in progress to make research necessary for Elven items instead of having them available by default from the start.

Fixes #13

### CODE REVIEW

File: Defs/RecipeDefs/LotRE_Recipes.xml [hunk 1]
```diff
@@ -4,10 +4,10 @@
 
    <!-- Without vegetable garden, Lembas bread is produced with lembas cornmeal. -->
    <RecipeDef>
-	<defName>LotRE_BakeElvishBread</defName>
```
Comment (jecrell) [inline Defs/RecipeDefs/LotRE_Recipes.xml:7]: Just a reminder of what I mentioned in the chatroom.
DefNames must not be changed.
Changing a DefName will cause existing games to crash and burn in a firey ball of deathmetal.

File: Defs/ResearchProjectDefs/LotRE_Research.xml [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    <label>Regal Elven Furniture</label>
+    <description>Furniture of such high quality it could only be Elven made:\n- Elven royal bed</description>
+    <baseCost>250</baseCost>
+    <prerequisites>
+      <li>LotRE_FurnitureOne</li>
+    </prerequisites>
+    <researchViewX>0.75</researchViewX>
+    <researchViewY>2.3</researchViewY>
+  </ResearchProjectDef>
+
```
Comment (jecrell) [inline Defs/ResearchProjectDefs/LotRE_Research.xml:146]: All these additions to Elven research look solid. I appreciate the attention to detail and mirroring of the Dwarves' research tab in style.

File: Defs/RecipeDefs/LotRE_Recipes.xml (hunk 1 above)
Comment (RodEsp) [reply Defs/RecipeDefs/LotRE_Recipes.xml:7]: Ok I think we're good here now. All the defNames should be the originals again.

**Comment by RodEsp** [comment]
Heya @jecrell should I add Recipes to research?
If not, I think this is done and ready for merging.

**Comment by jecrell** [comment]
Quick check -- did you add / delete anything for defs?

**Comment by RodEsp** [comment]
@jecrell nope I left all the `<defName>`s alone. The shields, armors, and weapons look like they changed because I rearranged them by research/power level using `<!-- Tier # -->` comments like the Dwarven ones are but all the defNames are the same.

**Comment by jecrell** [comment]
Great! I'll make a save and test it out to make sure nothing explodes. If
the red errors do not fly everywhere I'll upload it today.

**Comment by RodEsp** [comment]
Ok, now this should ACTUALLY be good.
😤

**Comment by RodEsp** [comment]
Alright, that's all the recipes.
The only thing I can see that's left is `LotRE_ElvenForestPlants.xml` and `LotRE_ElvenCorn.xml` but I don't know if those should be behind research? What do you think?

**Comment by jecrell** [comment]
Yeah don't worry about those. Plants shouldn't need research to plant --
just skill to manage. We might introduce a special backstory requirement
for them at a later date.

I'll look through it in about 8 hours and push after a test.

Also, unrelated, I've got a couple weeks left before I take off to the
states. What technologies/ are you working on over at Salesforce? I can get
ready while I've got a couple weeks.

**Comment by jecrell** [comment]
Everything looks good now. I have no errors on my end, and when technologies are fully researched, they appear to work correctly. I'll merge this and push it out to the public.
Lovely work on finding parallel phrases to match the Dwarven research, btw.

**Pull request merged**